### PR TITLE
[Banner] Add document for BannerView.

### LINKED ISSUE #7076: [Banner] Write usage documentation for the BannerView
This was filed as an internal issue. If you are a Googler, please visit [b/129977834](http://b/129977834) for more details.

---
#### Internal data
- Associated internal bug: [b/129977834](http://b/129977834)

### PR DESCRIPTION
**Pull request by fluxminds**
closes #7076

### CODE REVIEW

File: components/Banner/README.md [hunk 1]
```diff
@@ -1,8 +1,156 @@
...
+  - [Styling](#styling)
+  - [LayoutMargins](#layoutmargins)
+- [Examples](#examples)
+  - [Creating a banner view](#creating-a-banner-view)
+
+- - -
+
+## Overview
+
+`MDCBannerView` is a view that displays important, succinct message, and provides actions for users to address (or dismiss the banner). It requires a user action to be dismissed.
```
Comment (yarneo) [inline components/Banner/README.md]: nit: "an important"

File: components/Banner/README.md [hunk 2]
```diff
@@ -1,8 +1,156 @@
...
+
+`MDCBannerView` is currently a [beta component](../../contributing/beta_components.md). Therefore, clients that
+wish to use MDCBannerView in their app will need to manually specify the MaterialComponentsBeta target and our repository.
```
Comment (yarneo) [inline components/Banner/README.md]: nit: please stay consistent with the ` ` on class names and add it to MDCBannerView here as well.

Comment (yarneo) [reply components/Banner/README.md]: I'm not sure what it means to specify the target? do you mean: "wish to use MDCBannerView in their app will need to manually clone the repo and add the code to their project."

File: components/Banner/README.md [hunk 3]
```diff
@@ -1,8 +1,156 @@
...
+## Installation
+
+<!-- Extracted from docs/../../../docs/component-installation.md -->
+
+### Installation with CocoaPods
+
+Add the following to your `Podfile`:
+
+```bash
+pod 'MaterialComponents/Banner'
```
Comment (yarneo) [inline components/Banner/README.md]: MaterialComponentsBeta/

File: components/Banner/README.md [hunk 4]
```diff
@@ -1,8 +1,156 @@
...
+```
+
+### Importing
+
+To import the component:
+
+<!--<div class="material-code-render" markdown="1">-->
+#### Swift
+```swift
+import MaterialComponents.MaterialBanner
```
Comment (yarneo) [inline components/Banner/README.md]: MaterialComponentsBeta.

File: components/Banner/README.md [hunk 5]
```diff
@@ -1,8 +1,156 @@
...
+#import "MaterialBanner.h"
+```
+<!--</div>-->
+
+
+## Usage
+
+<!-- Extracted from docs/typical-use.md -->
+
+### Importing
```
Comment (yarneo) [inline components/Banner/README.md]: It looks like you are doing the Import step twice here

File: components/Banner/README.md [hunk 6]
```diff
@@ -1,8 +1,156 @@
...
+
+```objc
+#import "MaterialBanner.h"
+```
+
+<!--</div>-->
+
+### Appearance
+
+By default, the `MDCBannerView` is configured to display an image view, a text label and two buttons. To hide the image view on `MDCBannerView`, users can set the `hidden` property on `imageView` to be true. Similarly, users can hide a button on banner view by setting the `hidden` property on `trailingButton` to be true.
```
Comment (yarneo) [inline components/Banner/README.md]: "on the banner view"

File: components/Banner/README.md [hunk 7]
```diff
@@ -1,8 +1,156 @@
...
+
+### Styling
+
+By default, the `MDCBannerView` is configured to display items with black text and white background. To customize the color and style of the text, image view and buttons displayed on `MDCBannerView`, directly set relevant properties, such as `tintColor`, on `textLabel`, `imageView`, `leadingButton` and `trailingButton`.
```
Comment (yarneo) [inline components/Banner/README.md]: nit: By default, `MDCBannerView`...

Comment (yarneo) [reply components/Banner/README.md]: nit: and a white background

Comment (yarneo) [inline components/Banner/README.md]: nit: directly set the relevant properties

Comment (yarneo) [reply components/Banner/README.md]: nit: remove "on" from `textLabel`

File: components/Banner/README.md [hunk 8]
```diff
@@ -1,8 +1,156 @@
...
+
+### LayoutMargins
+
+`MDCBannerView` uses `layoutMargins` to manage the margins for elements on the banner view.
+
```
Comment (yarneo) [inline components/Banner/README.md:94]: have only one new line

**Review by yarneo — APPROVED** [review]
I provided comments on the main doc file, I'm expecting the changes to be done in the docs/*.md files and then regenerated.

LGTM % comments

Approving to unblock you once you resolve the comments. Thanks!

File: components/Banner/README.md (hunk 1 above)
Comment (fluxminds) [reply components/Banner/README.md]: Addressed in https://github.com/material-components/material-components-ios/pull/7291/commits/acc72d593483e098cb41c30814b599549b1eee99#diff-3006ac3fd7a404a379bdea38853b9383L34

File: components/Banner/README.md (hunk 2 above)
Comment (fluxminds) [reply components/Banner/README.md]: Addressed in https://github.com/material-components/material-components-ios/pull/7291/commits/acc72d593483e098cb41c30814b599549b1eee99#diff-3006ac3fd7a404a379bdea38853b9383L37

File: components/Banner/README.md (hunk 3 above)
Comment (fluxminds) [reply components/Banner/README.md]: Addressed in https://github.com/material-components/material-components-ios/pull/7291/commits/acc72d593483e098cb41c30814b599549b1eee99#diff-3006ac3fd7a404a379bdea38853b9383L37

File: components/Banner/README.md (hunk 4 above)
Comment (fluxminds) [reply components/Banner/README.md]: Addressed in 9200f91edf76b9615d52611989c4195512802841

File: components/Banner/README.md (hunk 5 above)
Comment (fluxminds) [reply components/Banner/README.md]: Addressed in acc72d593483e098cb41c30814b599549b1eee99

File: components/Banner/README.md (hunk 6 above)
Comment (fluxminds) [reply components/Banner/README.md]: Addressed in https://github.com/material-components/material-components-ios/pull/7291/commits/acc72d593483e098cb41c30814b599549b1eee99#diff-92b2aac05a9817deefbd21e797f875f3L22

File: components/Banner/README.md (hunk 7 above)
Comment (fluxminds) [reply components/Banner/README.md]: Addressed in https://github.com/material-components/material-components-ios/pull/7291/commits/acc72d593483e098cb41c30814b599549b1eee99#diff-92b2aac05a9817deefbd21e797f875f3L26

Comment (fluxminds) [reply components/Banner/README.md]: "on" is needed since tintColor will be set on `textLabel`.

File: components/Banner/README.md (hunk 8 above)
Comment (fluxminds) [reply components/Banner/README.md:94]: Addressed on acc72d593483e098cb41c30814b599549b1eee99

**Pull request merged**